File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func swapPairs(head *ListNode) *ListNode {
    dummy:=&ListNode{Val:0,Next:head}
    prev:=dummy
    for prev.Next!=nil && prev.Next.Next != nil{
        first:=prev.Next
        second:=first.Next
        first.Next=second.Next
        second.Next=first
        prev.Next=second
        prev=first
    }
    return dummy.Next
}

File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy=ListNode()
        dummy.next=head
        cur=dummy
        while cur.next and cur.next.next:
            temp=cur.next
            second=temp.next
            temp.next=second.next
            second.next=temp
            cur.next=second
            cur=cur.next.next
        return dummy.next